addons/mail/static/src/core/common/composer_actions.js

```diff
@@ -101,15 +101,7 @@ composerActionsRegistry
         sequenceQuick: 20,
     })
     .add("upload-files", {
-        condition: (component) => {
-            const thread = component.thread ?? component.message?.thread;
-            return (
-                !(
-                    thread?.channel_type === "whatsapp" &&
-                    component.props.composer.attachments.length > 0
-                ) && !component.props.composer.portalComment
-            );
-        },
+        condition: (component) => component.allowUpload,
```

Comment:
All the conditions are probably useful, you can't just replace them with the props.

Reply:
You should probably just try to add the props to the existing condition. This might fix your issue.